README.md

```diff
@@ -0,0 +1,109 @@
+1. **it took at least 6 weeks to get the first release on crates.io**
+   * development wasn't the fastest, and I claim one was slowed down due to too much manual testing.
+1. **there was no way to use the CI to test CLIs to actually interact with Google APIs**
+   * This was due to API usage being bound to a person, and these credentials were nothing you would want to have lying around in a public git repository.
+   * Not being able to test certain feature automatically and repeatedly takes time and reduces quality guarantees.
+1. **tests were minimal and most testing was like "if it compiles, it's good"**
+   * The _OP_ suffered from only having a few tests, and even though Rust only compiles decent quality software, by nature, certain reasoning was just in my head. _'Why is it doing this particular thing? It seems wrong'_ would be impossible to know. 
+   * manual testing is slow and error prone
+1. **Versions like 1.2.0+2019-08-23...` would additionally show the version of the Google API description, but is ignored by `cargo`** 
+   * This was done to discriminate the 'code' version from the version of the API it represents.
+   * As the `+` is ignored by `cargo`, to re-release a crate with a new version of the API, one would have to increment the patch level. However, that would force all crates to be re-released, even if their API version didn't change at all.
+   * This caused unnecessary spamming of `crates.io`, and the `+` should be a `-` to fix this.
+1. **The 'fields' parameter could not be used to have smaller response types**
+   * Some [Response Types](https://docs.rs/google-sheets4/1.0.10+20190625/google_sheets4/struct.Response.html) are huge, even though with the right `field` setting, one would only receive a fraction of the data. However, one would always have to allocate big structures to with most of the optional fields set to `None`.
+   * 💡Idea 💡: Can [serde(flatten)](https://serde.rs/field-attrs.html#flatten) be used to subdivide the possible field sets in the data structure? Probably it's not known which actual fields belong to each `field` argument.
+
+
+# Technology and Architecture Sketches
+
+Items mentioned below ideally create a link to one of the problems they slove, e.g. `🥵2` , the project goal they support, e.g `🛸3`, or the effective thing they build on (`🌈1`).
+
+## Toolchains
+
+Here is the anticipated tooling. What follows is the list of tools I would add and why.
+
+* **make** - repeats `🌈1` 
+  * I am a fan of simple makefiles, which catch dependencies between files and run a script to generate them. This served _OP_ extremely well.
+  * get parallelization for free, and make transparent which programs to call and how to get work done.
+  * the Makefile serves as hub keeping all commands one would run to interact with the project in any way.
```

Comment:
I said my piece about make earlier so I won't beat a dead horse, but I do agree that having a central place to see all the routine actions is important. I would just like to see that place be the default rust binary executed when somebody types `cargo run`, which by default would output a help page. I think rust provides all the functionality necessary and even the parallelism should be trivial if implemented with that in mind.

Reply:
I agree, as stated above, I will think `cargo run` and build scripts (possibly) to see how that could work and how it feels. (_nothing beats just typing a few commands and piping them together, sh is incredibly good in program orchestration_)